packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -0,0 +1,499 @@
+                        ""
+                    ),
+                }}
+            >
+                {this.props.children}
+            </Listbox>
+        );
+    }
+
+    /**
+     * Updates selection state and associated values
+     */
+    private updateSelection = (newSelection: ListboxItemProps[]): void => {
+        const newValue: string | string[] = this.getValueFromSelection(newSelection);
+        const newDisplayString: string = this.getFormattedDisplayString(newSelection);
+        this.setState({
+            selectedItems: newSelection,
+            value: newValue,
+            displayString: newDisplayString,
+        });
+        if (this.props.onValueChange) {
+            this.props.onValueChange(newValue, newSelection, newDisplayString);
+        }
+    };
+
+    /**
+     * Extracts values in the correct format (string in single select and string[] in multi-select) from an array of items.
+     */
+    private getValueFromSelection = (
+        newSelection: ListboxItemProps[]
+    ): string | string[] => {
+        if (this.props.multiselectable) {
+            return newSelection.map((thisItem: ListboxItemProps) => {
+                return thisItem.value;
+            });
+        } else {
+            if (newSelection.length === 0) {
+                return "";
+            }
+            return newSelection[0].value;
+        }
+    };
+
+    /**
+     * The default function that renders an unstyled content display
+     */
+    private defaultDisplayRenderFunction = (
+        props: SelectProps,
+        state: SelectState
+    ): React.ReactNode => {
+        if (props.multiselectable) {
+            return null;
+        }
+        return (
+            <Button
+                disabled={props.disabled}
+                aria-labelledby={props.labelledBy || null}
+                aria-haspopup={true}
+                aria-expanded={state.isMenuOpen}
+            >
+                {state.displayString}
+            </Button>
+        );
+    };
+
+    /**
+     * The default function that formats the display string generated based on selection.
+     * This implementation should match the default formatting a base html select control applies.
+     * Developers can provide an alternate formatter if desired.
+     */
+    private defaultDisplayStringFormatter = (
+        selectedOptions: ListboxItemProps[],
+        placeholder: string
+    ): string => {
+        const separator: string = ", ";
+        let returnValue: string = "";
+        if (selectedOptions.length > 0) {
+            selectedOptions.forEach((thisOption: ListboxItemProps) => {
+                if (returnValue.length > 0) {
+                    returnValue = returnValue + separator;
+                }
+                returnValue =
+                    returnValue +
+                    (thisOption.displayString === undefined
+                        ? thisOption.value
+                        : thisOption.displayString);
+            });
```

Comment:
for this method I believe you can just do:

```
const optionValues: string[] = selectedOptions.map(
    (selectedOption: ListboxItemProps): string => {
        return selectedOption.displayString || selectedOption.value;
    }
);

return selectOptions.length > 0 ? optionValues.join(", ") : placeholder;
```

Reply:
yep, cleaner